src/processor.cpp

```diff
@@ -0,0 +1,401 @@
+    } else if( processor_id->examine != converter ) {
+        debugmsg( "Furniture \"%s\" can't act as a processor because the assigned iexamine function is not converter, but it has \"%s\" associated.",
+                  processor_tid.c_str(), id.c_str() );
+    }
+    for( const auto &elem : processes ) {
+        if( !elem.is_valid() ) {
+            debugmsg( "Invalid process \"%s\" in \"%s\".", elem.c_str(), id.c_str() );
+        }
+    }
+}
+
+void processor_data::load( JsonObject &jo, const std::string &src )
+{
+    processors_data.load( jo, src );
+}
+
+void processor_data::check()
+{
+    processors_data.check();
+}
+
+void processor_data::reset()
+{
+    processors_data.reset();
+}
+
+//process namespace
+
+namespace
+{
+
+struct process_data;
+using process_id = string_id<process_data>;
+
+struct process_data {
+    process_data():
+        was_loaded( false ) {};
+
+    process_id id;
+    std::string name;
+    auto components;
+    auto output;
+    int fuel_intake;
+    int duration;
+    //TODO: Skill used
+
+    void load( JsonObject &jo, const std::string &src );
+    void check() const;
+
+};
+
+process_id get_process_id( std::string id )
+{
+    return process_id( id );
+}
+
+generic_factory<process_data> processes_data( "process type", "id" );
+
+}
+
+void process_data::load( JsonObject &jo, const std::string & )
+{
+    mandatory( jo, was_loaded, "name", name );
+    //TODO: modify to allow lists of components
+    mandatory( jo, was_loaded, "components", components,  string_id_reader<itype_id> {});
+    mandatory( jo, was_loaded, "output", output );
+    mandatory( jo, was_loaded, "fuel_intake", fuel_intake );
+    mandatory( jo, was_loaded, "duration", duration );
+}
+
+void process_data::check() const
+{
+    const process_id proc_id( id.str() );
+
+    if( !proc_id.is_valid() ) {
+        debugmsg( "There is no process with the name %s", id.c_str() );
+    if( !components.is_valid() ) {
+        debugmsg( "Invalid components \"%s\" in \"%s\".", components.c_str(), id.c_str() );
+    }
+    if( !output.is_valid() ) {
+        debugmsg( "Invalid output \"%s\" in \"%s\".", output.c_str(), id.c_str() );
+    }
+    if( !fuel_intake.is_valid() ) {
+        debugmsg( "Invalid fuel intake \"%s\" in \"%s\".", fuel_intake.c_str(), id.c_str() );
+    }
+    if( !duration.is_valid() ) {
+        debugmsg( "Invalid duration \"%s\" in \"%s\".", duration.c_str(), id.c_str() );
+    }
+}
+
+void process_data::load( JsonObject &jo, const std::string &src )
+{
+    processes_data.load( jo, src );
+}
+
+void process_data::check()
+{
+    processes_data.check();
+}
+
+void process_data::reset()
+{
+    processes_data.reset();
+}
+
+void interact_with_processor( const tripoint &examp, player &p )
+{
+    //Get processor id at examp
+    processor_id pid = get_processor_id( examp );
+    //Do I need the debugmsg-es here or do they get checked in process_data::load?
```

Comment:
Yes, the debugmsgs are useful here in case someone badly specifies the processor `iexamine` for a terrain/furniture that doesn't have a processor set.